nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/CandidateRepository.java

```diff
@@ -122,13 +121,13 @@ public CandidateDao create(DbCandidate dbCandidate, List<DbApprovalStatus> appro
         return candidateTable.getItem(candidate);
     }
 
-    public void updateCandidate(UUID identifier, CandidateDao candidateDao, List<DbApprovalStatus> approvals) {
+    public void updateCandidate(CandidateDao candidateDao, List<DbApprovalStatus> approvals) {
         var approvalMap = approvals.stream()
-                              .map(approval -> mapToApprovalStatusDao(identifier, approval))
+                              .map(approval -> mapToApprovalStatusDao(candidateDao.identifier(), approval))
                               .collect(toMap(approvalStatusDao -> approvalStatusDao.approvalStatus().institutionId(),
                                              Function.identity()));
+        removeOldApprovals(candidateDao.identifier());
```

Comment:
This is the most significant change

Reply:
It might not be wise to split up `updateCandidate` in two transactions like I have done now: 
1. removing old approvals
2. updating candidate with new approvals

There is no rollback of the first transaction if the second one fails. I'll see if i can refactor this so both are done in one transaction.